lib/intern.js

```diff
@@ -0,0 +1,160 @@
+const _ = require('lodash')
+const Uuid = require('node-uuid')
+
+
+function ensureId(ent) {
+  if (undefined !== ent.id$) {
+    return ent.id$
+  }
+
+  return Uuid()
+}
+
+
+function isNew(ent) {
+  return !('id' in ent)
+}
+
+
+function escapeStr(input) {
+  const str = '' + input
+
+  return str.replace(/[\0\b\t\x08\x09\x1a\n\r"'\\\%]/g, function (char) {
+    switch (char) {
+      case '\0':
+        return '\\0'
+      case '\x08':
+        return '\\b'
+      case '\b':
+        return '\\b'
+      case '\x09':
+        return '\\t'
+      case '\t':
+        return '\\t'
+      case '\x1a':
+        return '\\z'
+      case '\n':
+        return '\\n'
+      case '\r':
+        return '\\r'
+      case '\"':
+      case "'":
+      case '\\':
+      case '%':
+        return '\\' + char
+    }
+  })
+}
+
+
+function maybeJsonField(field, row, schema) {
+  /* NOTE:
+  Source: https://www.json.org/json-en.html
+
+  > JSON is built on two structures:
+  >
+  > A collection of name/value pairs. In various languages, this is realized
+  > as an object, record, struct, dictionary, hash table, keyed list, or
+  > associative array.
+  > 
+  > An ordered list of values. In most languages, this is realized as an
+  > array, vector, list, or sequence.
+  */
+  const mayTryParse = mayStoreJson(field, row, schema) &&
```

Comment:
are we testing each field value to see if it might be JSON? - that's gonna be really slow
take the value as is and only try to JSON parse if the options indicate it is an object etc